tests/test_webdav.py

```diff
@@ -0,0 +1,77 @@
+import subprocess
+import pytest
+import tempfile
+from webdav4.client import Client, HTTPError
+from conftest import webdav_server, generate_random_file
+
+
+@pytest.mark.timeout(30)
+def test_webdav4_client_list(start_service):
+    with webdav_server():
+        client = Client(start_service['url'] + "/webdav/lst")
+        res = client.ls("", detail=True)
+        assert len(res) == 1
+        assert res[0]['href'] == "/webdav/lst/users/"
+        assert res[0]['type'] == "directory"
+
+
+@pytest.mark.timeout(30)
+def test_webdav4_client_upload_denied(start_service):
+    with webdav_server():
+        client = Client(start_service['url'] + "/webdav/lst")
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            local_file = f"{tmpdir}/local-file"
+            generate_random_file(local_file, 1*(1024**2))
+
+            with pytest.raises(HTTPError):
+                try:
+                    client.upload_file(
+                        local_file,
+                        'uploaded-file',
+                        chunk_size=1024**2,
+                    )
+                except HTTPError as e:
+                    assert "Missing rights to write in" in e.__str__()
+                    raise e
+
+
+@pytest.mark.timeout(30)
+def test_webdav4_client_upload_valid(start_service):
+    with webdav_server():
+        client = Client(start_service['url'] + "/webdav/lst")
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            local_file = f"{tmpdir}/local-file"
+            generate_random_file(local_file, 100*(1024**2))
+
+            client.upload_file(
+                local_file,
+                'users/anonymous/uploaded-file',
+                chunk_size=1024**2,
```

Comment:
is there a way to verify that it is indeed chunked?

Reply:
No, I do not hink so, bu I checked during development in the `downloadservice` that requests where indeed using chunk encoding.

Reply:
could you add a "# TODO:" comment then so we can think to add it in the future if needed?